### PR TITLE
Adding Changelog for v0.3.0

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds a changelog for v0.3.0. Approving this PR indicates that you approve of a v0.3.0 release based on this commit.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @bowei @danehans @hbagdi @jpeach

### CODE REVIEW

**Comment by robscott** [comment]
Adding a hold on this until at least 2 maintainers LGTM this to ensure we have consensus on this release.

/hold

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,55 @@
...
+
+API Version: v1alpha1
+
+### API changes
+
+#### Gateway
+- The `NoSuchGatewayClass` status reason has been deprecated.
+  [#635](https://github.com/kubernetes-sigs/gateway-api/pull/635)
+
+#### HTTPRoute
```
Comment (danehans) [inline CHANGELOG.md:21]: @robscott should we indicate that path match is now a pointer?

xref: https://github.com/kubernetes-sigs/gateway-api/pull/564

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,55 @@
...
 - [v0.1.0-rc2](#v010-rc2)
 - [v0.1.0-rc1](#v010-rc1)
 
+## v0.3.0
+
+API Version: v1alpha1
+
+### API changes
+
+#### Gateway
```
Comment (danehans) [inline CHANGELOG.md:17]: @robscott should we indicate that route selectors are now a pointer?

xref: https://github.com/kubernetes-sigs/gateway-api/commit/dc82c09b08b816fb786004eb6af58ac993c2bb34

**Pull request merged**